dts/bindings/ethernet/st,stm32-ethernet-common.yaml

```diff
@@ -18,3 +18,5 @@ properties:
     required: true
   pinctrl-names:
     required: true
+  phy-connection-type:
+    default: "rmii"
```

Comment:
See https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#dt-bindings-default-rules, but I'm not sure default really matches constraints here.

Reply:
it is done the same as in dts/bindings/ethernet/atmel,gmac-common.yaml. 
By the way, this helps keeping the compatibility, as that was the default for the Kconfigs (you had to select a Kconfig if you wanted something else than rmii).